lib/Graph/Graph.cpp

```diff
@@ -62,10 +62,22 @@ void Module::strip() {
 }
 
 void Module::clear() {
-  eraseFunctions();
-
   for (auto it = constants_.begin(), e = constants_.end(); it != e; it++) {
     Constant *v = *it;
+
+    // Log constant deletion in all its user functions.
+    FunctionList userFuncs;
+    for (auto use : v->getUsers()) {
+      auto userFunc = use.getUser()->getParent();
+      if (std::find(userFuncs.begin(), userFuncs.end(), userFunc) ==
```

Comment:
Consider performing the logging of node deletion in this loop itself. To avoid double deletion, you can keep a `std::unordered_set` or equivalent of `Function*` and check if a function is already there before logging. I think that will be faster than `std::find` in a list.